cluster/src/dunit/scala/io/snappydata/cluster/QueryRoutingDUnitSecureTest.scala

```diff
@@ -0,0 +1,402 @@
+      case x: SQLException if x.getSQLState.equals("42500") => // ignore
+      case t: Throwable => throw t
+    }
+    insertRows2(2000, serverHostPort, tableName, jdbcUser2, jdbcUser2)
+
+    // (1 to 5).foreach(d => query())
+    try {
+      query1(serverHostPort, jdbcUser2 + "." + tableName, jdbcUser1, jdbcUser1)
+      assert(false) // fail
+    } catch {
+      case x: SQLException if x.getSQLState.equals("42502") => // ignore
+      case t: Throwable => throw t
+    }
+    query1(serverHostPort, tableName, jdbcUser2, jdbcUser2)
+
+    try {
+      dropTable(serverHostPort, jdbcUser2 + "." + tableName, jdbcUser1, jdbcUser1)
+      assert(false) // fail
+    } catch {
+      case x: SQLException if x.getSQLState.equals("42507") => // ignore
+      case t: Throwable => throw t
+    }
+    dropTable(serverHostPort, tableName, jdbcUser2, jdbcUser2)
+  }
+
+  def testRowTableRouting(): Unit = {
+    val jdbcUser3 = "gemfire3"
+    val jdbcUser4 = "gemfire4"
+    val serverHostPort = AvailablePortHelper.getRandomAvailableTCPPort
+    vm2.invoke(classOf[ClusterManagerTestBase], "startNetServer", serverHostPort)
+    // scalastyle:off println
+    println(s"network server started at $serverHostPort")
+    // scalastyle:on println
+
+    val tableName = "order_line_row"
+
+    try {
+      createTable2(serverHostPort, jdbcUser3 + "." + tableName, jdbcUser4, jdbcUser4)
+      assert(false) // fail
+    } catch {
+      case x: SQLException if x.getSQLState.equals("42507") ||
+          x.getSQLState.equals("42508") => // ignore
+      case t: Throwable => throw t
+    }
+    createTable2(serverHostPort, tableName, jdbcUser3, jdbcUser3)
+
+    try {
+      insertRows3(200, serverHostPort, jdbcUser3 + "." + tableName, jdbcUser4, jdbcUser4)
+      assert(false) // fail
+    } catch {
+      case x: BatchUpdateException => // ignore
+      case t: Throwable => throw t
+    }
+    insertRows3(200, serverHostPort, tableName, jdbcUser3, jdbcUser3)
+
+    try {
+      insertRows4(200, serverHostPort, jdbcUser3 + "." + tableName, jdbcUser4, jdbcUser4)
+      assert(false) // fail
+    } catch {
+      case x: SQLException if x.getSQLState.equals("42500") => // ignore
+      case t: Throwable => throw t
+    }
+    insertRows4(200, serverHostPort, tableName, jdbcUser3, jdbcUser3)
+
+    // (1 to 5).foreach(d => query())
+    try {
+      query2(serverHostPort, jdbcUser3 + "." + tableName, jdbcUser4, jdbcUser4)
+      assert(false) // fail
+    } catch {
+      case x: SQLException if x.getSQLState.equals("42502") => // ignore
+      case t: Throwable => throw t
+    }
+    query2(serverHostPort, tableName, jdbcUser3, jdbcUser3)
+
+    try {
+      dropTable(serverHostPort, jdbcUser3 + "." + tableName, jdbcUser4, jdbcUser4)
+      assert(false) // fail
+    } catch {
+      case x: SQLException if x.getSQLState.equals("42502") => // ignore
+      case t: Throwable => throw t
+    }
+    dropTable(serverHostPort, tableName, jdbcUser3, jdbcUser3)
+  }
+}
```

Comment:
Some functions look common to both ConcurrentQueryRoutingDUnitSecureTest and QueryRoutingDUnitSecureTest  (insertRows1, insertRows2, verifyQuery1 etc.). Can we keep these in one class/object and instead of repeating/copying in both tests. If there is an issue in these functions we don't have to fix it at multiple places.